packages/vscode-extension/src/commandHandlers.ts

```diff
@@ -74,9 +73,12 @@ export async function saveConnectionAndDisplayConnectionResult(
     connection,
     password,
   );
+  const errorDetail = result?.error?.friendlyMessage;
 
-  if (!result.success && result.retriable) {
```

Comment:
After a discussion with Greg: we should always be offering the option to store the connection when credentials or url are invalid.

The difference here was that if I was trying with an invalid url, I would be offered to store the connection regardless, whereas if I was trying with invalid credentials but the url was correct, it would not offer me to store.